tests/driver_adxl345/main.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2016 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       ADXL345 test application
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "adxl345.h"
+#include "adxl345_params.h"
+#include "xtimer.h"
+
+#define SLEEP_DELAY 100 * 1000U
+
+static adxl345_t dev;
+
+int main(void)
+{
+    adxl345_data_t data;
+
+    puts("ADXL345 test application");
+    printf("Initializing ADXL345 accelerometer at I2C_DEV(%i)... ",
+           adxl345_params->i2c);
+
+    if (adxl345_init(&dev, adxl345_params) == ADXL345_OK) {
```

Comment:
I would prefer testing the error return code first (NOI2C,  NODEV) and display the corresponding reason.